scripts/jsign.mjs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {spawn} from 'child_process';
+import {resolve} from 'path';
+
+/**
+ * Get the full path of jsign.jar.
+ * @returns {string} The full path similar to <.../jsign.jar>.
+ */
+function getJsignJarPath() {
+  return resolve('third_party', 'jsign', 'jsign-4.0.jar');
+}
+
+/**
+ * Run jsign.jar according to the corresponding options targeting fileToSign.
+ * @param {string} fileToSign The path string of a file to be signed.
+ * @param {string[]} options The options to be passed to jsign. see https://ebourg.github.io/jsign/
+ * @returns {Promise<number>} A promise containing the exit code of jsign.
+ */
+export default async function jsign(fileToSign, options) {
```

Comment:
Hm, I defer to @fortuna here but generally our pattern has been to not spawn processes from node scripts, at least not yet. that's more of an "action"-y kind of thing

if you wanted to be more in line with our current patterns this would become `get_jsign_jar_path` and log the path to the console for use in an action.sh. (preferrably `electron/build`) see `scripts/get_version.mjs` for an example.

I want to move everything toward node as well but feel we should maintain a distinction between utility scripts and action(s), too.

Reply:
I see. My idea is to merge utility and actions into one js file, so other js files can either call it as an action or use it as a helper method/class. If we do need so, we can put them into different subfolders like `utility` or `action`.

Reply:
actions are generally located where they're run from - the path you specify when running the action points directly to its location - so a top-level action folder would break this

I think it becomes a bit unclear what's going to happen when you call it from another file, then. Will it spin up a new process, or no? Do I get a "process object" I can manipulate? maybe it comes down to that clarity, basically, since we're doing what you describe in the other scripts, but generally the MO so far of the node scripts so far is to pull data for the master bash process - deterministic inputs and outputs with no side effects

Reply:
Also, what's the rationale for breaking this out, instead of making it a part of "sign_windows_executable"?

Reply:
I agree to resolve `jsign.jar` to the absolute path using something like `__dirname` so we don't need to worry about the path things as long as we don't change the relative folder structure.

`jsign.mjs` will create a new sub-process, wait until it exists, and return the exit code to the caller. My underlying rationale for separating the files is:

1. Single responsibility principle: `sign_windows_executable` would be focus on the signing logic; and `jsign` focus on invoking `jsign.jar` and returns the exit code.
2. In case anyone else would need to call `jsign` (similar to an npm package, e.g. [openssl-nodejs](https://www.npmjs.com/package/openssl-nodejs)).

Reply:
yeah, in the `main` when you run it as a script? I think that feels right